### ISSUE TITLE
Test case of `DockerSuite.TestEventsOOMDisableFalse` is not reliable on AArch64

### ISSUE THREAD

**Issue by arm64b** [moby/moby#36090]
**Description**
On AArch64 platform, we have a failed test case as:
```
----------------------------------------------------------------------
FAIL: docker_cli_events_unix_test.go:81: DockerSuite.TestEventsOOMDisableTrue

docker_cli_events_unix_test.go:120:
    c.Fatalf("container should be still running but it's not")
... Error: container should be still running but it's not
----------------------------------------------------------------------
```

Since I am trying to make all the CI tests passed on AArch64 for our own testing pipeline, consequently I need to triage this failure. After some investigation, I found this test case has been skipped by all the archs such as amd64/powerpc/s390 like:
` DockerSuite.TestEventsOOMDisableTrue (unmatched requirement swapMemorySupport)`

But I still want to know what the reason is, so I have below experiment:
Open 2 terminals A and B, On terminal A, I run below command:
`docker events`
On terminal B, first I run this command:
`docker run --oom-kill-disable=true -m 10MB -v "$PWD:/tmp" busybox /tmp/oom.sh`
The output of this command on terminal A is:
```
2018-01-23T08:30:30.697759420Z container create 66bfd6dcc99e984fe8a55dc9227ceaa977f9bcdaa6901f50ee2dd67baf445af2 (image=busybox, name=agitated_allen)
2018-01-23T08:30:30.700100323Z container attach 66bfd6dcc99e984fe8a55dc9227ceaa977f9bcdaa6901f50ee2dd67baf445af2 (image=busybox, name=agitated_allen)
2018-01-23T08:30:30.899210809Z network connect 6495e6a0a7b06fbd733fae6d6a8dcb915a6f595309b7d5f6888fcc63a4aa7ca5 (container=66bfd6dcc99e984fe8a55dc9227ceaa977f9bcdaa6901f50ee2dd67baf445af2, name=bridge, type=bridge)
2018-01-23T08:30:32.074005698Z container start 66bfd6dcc99e984fe8a55dc9227ceaa977f9bcdaa6901f50ee2dd67baf445af2 (image=busybox, name=agitated_allen)
2018-01-23T08:30:32.124343251Z container oom 66bfd6dcc99e984fe8a55dc9227ceaa977f9bcdaa6901f50ee2dd67baf445af2 (image=busybox, name=agitated_allen)
2018-01-23T08:30:32.361344231Z container die 66bfd6dcc99e984fe8a55dc9227ceaa977f9bcdaa6901f50ee2dd67baf445af2 (exitCode=137, image=busybox, name=agitated_allen)
2018-01-23T08:30:33.332313427Z network disconnect 6495e6a0a7b06fbd733fae6d6a8dcb915a6f595309b7d5f6888fcc63a4aa7ca5 (container=66bfd6dcc99e984fe8a55dc9227ceaa977f9bcdaa6901f50ee2dd67baf445af2, name=bridge, type=bridge)
```

Second, I run below command on terminal B:
`docker run --oom-kill-disable=false -m 10MB -v "$PWD:/tmp" busybox /tmp/oom.sh`

The output on terminal A is:
```
2018-01-23T08:31:46.452001318Z container create 6a8d251aae8431fd6bfddbe51f4fbc1121731371a8761683a42ad754625d43d3 (image=busybox, name=amazing_torvalds)
2018-01-23T08:31:46.454747938Z container attach 6a8d251aae8431fd6bfddbe51f4fbc1121731371a8761683a42ad754625d43d3 (image=busybox, name=amazing_torvalds)
2018-01-23T08:31:46.664136536Z network connect 6495e6a0a7b06fbd733fae6d6a8dcb915a6f595309b7d5f6888fcc63a4aa7ca5 (container=6a8d251aae8431fd6bfddbe51f4fbc1121731371a8761683a42ad754625d43d3, name=bridge, type=bridge)
2018-01-23T08:31:47.619747130Z container start 6a8d251aae8431fd6bfddbe51f4fbc1121731371a8761683a42ad754625d43d3 (image=busybox, name=amazing_torvalds)
2018-01-23T08:31:47.670139196Z container oom 6a8d251aae8431fd6bfddbe51f4fbc1121731371a8761683a42ad754625d43d3 (image=busybox, name=amazing_torvalds)
2018-01-23T08:31:47.900055686Z container die 6a8d251aae8431fd6bfddbe51f4fbc1121731371a8761683a42ad754625d43d3 (exitCode=137, image=busybox, name=amazing_torvalds)
2018-01-23T08:31:48.880340263Z network disconnect 6495e6a0a7b06fbd733fae6d6a8dcb915a6f595309b7d5f6888fcc63a4aa7ca5 (container=6a8d251aae8431fd6bfddbe51f4fbc1121731371a8761683a42ad754625d43d3, name=bridge, type=bridge)
```

So we can see that no matter we enable the `--oom-kill-disable` or not, the behavior to run the same container process is the same.

**Steps to reproduce the issue:**
1.
2.
3.

**Describe the results you received:**
Same behavior between the 2 docker container process with `--oom-kill-disable` enable or not

**Describe the results you expected:**
The container process with `--oom-kill-disable=true` should still run until timeout occurs in terms of the test case.

**Additional information you deem important (e.g. issue happens only occasionally):**

**Output of `docker version`:**

```
Client:
 Version:      unknown-version
 API version:  1.34
 Go version:   go1.9.2
 Git commit:   unknown-commit
 Built:        unknown-buildtime
 OS/Arch:      linux/arm64

Server:
 Version:      dev
 API version:  1.36 (minimum version 1.12)
 Go version:   go1.9.2
 Git commit:   ef3988a
 Built:        Thu Jan 18 02:24:13 2018
 OS/Arch:      linux/arm64
 Experimental: true
```

**Output of `docker info`:**

```
Containers: 73
 Running: 1
 Paused: 0
 Stopped: 72
Images: 321
Server Version: dev
Storage Driver: overlay2
 Backing Filesystem: extfs
 Supports d_type: true
 Native Overlay Diff: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge host ipvlan macvlan null overlay
 Log: awslogs fluentd gcplogs gelf journald json-file logentries splunk syslog
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: 89623f28b87a6004d4b785663257362d1658a729
runc version: b2567b37d7b75eb4cf325b77297b140ea686ce8f
init version: 949e6fa
Security Options:
 seccomp
  Profile: default
Kernel Version: 4.9.20
Operating System: Ubuntu 16.04.3 LTS
OSType: linux
Architecture: aarch64
CPUs: 64
Total Memory: 125.6GiB
Name: entos-huawei-d05
ID: X4QW:BJJZ:3MVW:GKDT:7YHJ:NNTZ:GV24:JI3Y:CNIL:CXGQ:Y2FP:UJEM
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Username: arm64b
Registry: https://index.docker.io/v1/
Labels:
Experimental: true
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**

**Comment by arm64b** [comment]
The final goal for us is to pass/skip this test case.  Feel free to give your comments, or my suggestion is we add below statement to skip it on arm64 either:

`testRequires(c, DaemonIsLinux, oomControl, memoryLimitSupport, NotArm, NotArm64, swapMemorySupport)`

Any comments?

**Comment by arm64b** [comment]
Has been fixed by PR #36201, so we can close this issue
